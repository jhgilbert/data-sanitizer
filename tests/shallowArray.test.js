const { sanitizeFields } = require("..")

test('sanitizes a shallow array', () => {
  let data = [
    { id: 1, name: 'Jen Gilbert' },
    { id: 2, name: 'Derek Reeve' }
  ]

  const sanitationResult = sanitizeFields({ data, fieldNames: ['id'] })
  expect(sanitationResult).toMatchSnapshot()
})