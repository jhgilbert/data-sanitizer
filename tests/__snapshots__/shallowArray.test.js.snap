// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitizes a shallow array 1`] = `
Object {
  "data": Array [
    Object {
      "id": "[SANITIZED]",
      "name": "Jen Gilbert",
    },
    Object {
      "id": "[SANITIZED]",
      "name": "Derek Reeve",
    },
  ],
  "removedValues": Object {
    "id": Array [
      1,
      2,
    ],
  },
}
`;
